datadog/fwprovider/resource_datadog_apm_retention_filter.go

```diff
@@ -132,12 +133,73 @@ func (r *ApmRetentionFilterResource) Read(ctx context.Context, request resource.
 	response.Diagnostics.Append(response.State.Set(ctx, &state)...)
 }
 
+type CommonRequest struct {
+	Diagnostics diag.Diagnostics
+	State       *tfsdk.State
+}
+
+func NewCommonRequest(diag diag.Diagnostics, state *tfsdk.State) CommonRequest {
+	return CommonRequest{
+		Diagnostics: diag,
+		State:       state,
+	}
+}
+
+func (r *ApmRetentionFilterResource) getAndUpdate(state *ApmRetentionFilterModel, ctx context.Context, response CommonRequest) {
+	resp, _, err := r.Api.ListApmRetentionFilters(r.Auth)
+	if err != nil {
+		return
+	}
+	if err := utils.CheckForUnparsed(resp); err != nil {
+		return
+	}
+	var id string
+	var filterName string
+	for _, rfa := range resp.Data {
+		if string(rfa.Attributes.GetFilterType()) == state.FilterType.ValueString() {
+			state.ID = types.StringValue(rfa.Id)
+			id = rfa.Id
+			filterName = rfa.Attributes.GetName()
+			break
+		}
+	}
+
+	body, diags := r.buildApmRetentionFilterUpdateRequestBody(ctx, state)
+	body.Data.Attributes.SetName(filterName)
+
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	apmRetentionFilterMutex.Lock()
+	defer apmRetentionFilterMutex.Unlock()
+
+	respUpdate, _, err := r.Api.UpdateApmRetentionFilter(r.Auth, id, *body)
+	if err != nil {
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error dd retention filter"))
+		return
+	}
+	if err := utils.CheckForUnparsed(resp); err != nil {
+		response.Diagnostics.AddError("response contains unparsedObject", err.Error())
+		return
+	}
+	r.updateState(ctx, state, &respUpdate)
+
+	// Save data into Terraform state
+	response.Diagnostics.Append(response.State.Set(ctx, &state)...)
+}
+
 func (r *ApmRetentionFilterResource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
 	var state ApmRetentionFilterModel
 	response.Diagnostics.Append(request.Plan.Get(ctx, &state)...)
 	if response.Diagnostics.HasError() {
 		return
 	}
+	if state.FilterType.ValueString() != "spans-sampling-processor" {
```

Comment:
Can default retention rules not have  type `spans-sampling-processor `? I see Synthetics Default rule has definition:
```
                {
                    "id": "",
                    "name": "Synthetics Default",
                    "enabled": false,
                    "meta": {
                        "description": "Get started with Synthetic Monitoring",
                        "tags": [
                            "dd.default:true",
                            "dd.feature:synthetics"
                        ]
                    },
                    "type": "spans-sampling-processor"
                }
```